clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -156,7 +177,8 @@ else if (LZ4.name.equals(name))
     private static class LZ4Constructors {
         static final MethodHandle INPUT = findConstructor(
                 "org.apache.kafka.common.record.KafkaLZ4BlockInputStream",
```

Comment:
We followed the same approach for Snappy and LZ4 to deal with the possibility that the libraries are not in the classpath because it's not supported in a given platform. However, we are using Kafka classes for the block input stream and block output stream. As such, we can probably reference the constructor directly. We won't invoke the constructor unless LZ4 is configured and it's OK to fail in that case.

Does that make sense?